core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -467,6 +471,22 @@ public <T> ServerBuilder childChannelOption(ChannelOption<T> option, T value) {
         return this;
     }
 
+    /**
+     * The {@link Consumer} that customizes the Netty {@link ChannelPipeline}.
+     * This customizer is no-op by default.
+     *
+     * <p>Note that usage of this customizer is an advanced
+     * feature and may produce unintended side effects, including complete
+     * breakdown. It is not recommended if you are not familiar with
+     * Armeria and Netty internals.
+     */
+    public ServerBuilder channelPipelineCustomizer(
+            Consumer<? super ChannelPipeline> channelPipelineCustomizer) {
```

Comment:
Should we also add `ClientFactoryBuilder.channelPipelineCustomizer(Consumer) symmetrically?

Reply:
According to the conversation of #4260, shouldn't we add `ClientFactoryBuilder.channelPipelineCustomizer(Consumer)` until more users want this feature?
This PR provides a feature for server-side.
I suppose we should add this kind of feature for client-side in other issue.

Reply:
> According to the conversation of https://github.com/line/armeria/pull/4260, shouldn't we add ClientFactoryBuilder.channelPipelineCustomizer(Consumer) until more users want this feature?

We have an internal conversation and came to the conclusion that support `ClientFactoryBuilder.channelPipelineCustomizer(Consumer)` for consistency since we are adding `ServerBuilder.channelPipelineCustomizer(Consumer)`.

>  I suppose we should add this kind of feature for client-side in other issue.

👍 Sounds good.